File: 0471-12-3264-final-array-state-after-k-multiplication-operations-i/0471-12-3264-final-array-state-after-k-multiplication-operations-i.cpp
class Solution {
public:
    vector<int> getFinalState(vector<int>& nums, int k, int multiplier) {
        set<pair<int,int>>st;
        for(int i = 0 ; i < nums.size();i++)st.insert({nums[i],i});
        while(k--){
           auto it = *st.begin();
           st.erase(it);
           int idx = it.second;
           nums[idx]*=multiplier;
           st.insert({nums[idx],idx});
        }
        return nums;
    }
};

auto init = []() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    cout.tie(nullptr);
    return "MaTb3aa";
}();

File: 0471-12-3264-final-array-state-after-k-multiplication-operations-i/0471-12-3264-final-array-state-after-k-multiplication-operations-i.py
from sortedcontainers import SortedSet

class Solution:
    def getFinalState(self, nums, k, multiplier):
        st = SortedSet((num, i) for i, num in enumerate(nums))

        for _ in range(k):
            val, idx = st.pop(0)  
            nums[idx] *= multiplier
            st.add((nums[idx], idx)) 

        return nums
